### ISSUE TITLE
linter: `eslint/no-inner-declarations` does not report for `var` in for loop initializer

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#11634]
### What version of Oxlint are you using?

Latest main

### What happened?

The following cases are not considered an error under `eslint/no-inner-declarations` rule:

```js
for (var x in {}) {}
for (var x of []) {}
for (var x; x < 10; x++) {}
```

ESLint considers all these to be errors ([ESLint playground](https://eslint.org/play/#eyJ0ZXh0IjoiZm9yICh2YXIgeCBpbiB7fSkge31cbmZvciAodmFyIHggb2YgW10pIHt9XG5mb3IgKHZhciB4OyB4IDwgMTA7IHgrKykge31cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnsibm8taW5uZXItZGVjbGFyYXRpb25zIjpbImVycm9yIiwiYm90aCJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=)). Is there a reason why we don't?

**Comment by camc314** [comment]
@overlookmotel is this fixed by #11632 ?

I assigned to you. Do you mind closing if this is fixed? or assigning to me if it still needs more work? thank you

**Comment by overlookmotel** [comment]
No, #11632 fixed this case:

```js
for (const x in {}) var y = 1;
//                  ^^^^^ error
```

This issue is about a different case:

```js
for (var x in {}) {}
//   ^^^^^ error?
```

The question is: Should this 2nd case be an error or not?

ESLint makes it an error ([ESLint playground](https://eslint.org/play/#eyJ0ZXh0IjoiZm9yICh2YXIgeCBpbiB7fSkge31cbmZvciAodmFyIHggb2YgW10pIHt9XG5mb3IgKHZhciB4OyB4IDwgMTA7IHgrKykge31cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnsibm8taW5uZXItZGVjbGFyYXRpb25zIjpbImVycm9yIiwiYm90aCJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=)), so I'd assume it should be. But we have code that specifically looks for this pattern and makes it *not* an error, so at someone decided that's how it should be at some point. I'm not sure why.

If we decide that it *should* be an error, I'll be happy to fix it. I just don't know the linter well, so I thought I'd check first.

**Comment by camc314** [comment]
yes, it should be an error imo

```
 node
Welcome to Node.js v22.15.0.
Type ".help" for more information.
> for (var k = 123;false;) {}
undefined
> k
123
> 
```

It doesn't make sense that you can access `k` after the for loop scope has exited

**Comment by camc314** [comment]
To add, this behaviour has been there since the rule was originally added. I don't think it was deliberate are there were no tests for this behaviour.

https://github.com/oxc-project/oxc/commit/1aa112917a20a4b4c0ab31b89b7c05954b9a182a

**Comment by overlookmotel** [comment]
Fixed in #11692.
